roachtest: update fully decommission step in decommission/mixed-versions

renatolabs opened the pull request:
Previously, the fully decommissioning of a node would happen on a `AfterUpgradeFinalized` hook. However, that was not safe: a mutator could add a step after that hook and it would fail to run as it could try to connect to the node that was just decommissioned.

Since the full decommissioning was only happening after the upgrade to the `current` version, it is sufficient (and simpler) to run that step outside the mixedversion framework, after the `Run` function returns.

This PR also fixes the `decommission` commands to always talk to the storage cluster, to account for multi-tenant deployments.

Fixes: #131550

Release note: None

Commit:
roachtest: update fully decommission step in decommission/mixed-versions

Previously, the fully decommissioning of a node would happen on a
`AfterUpgradeFinalized` hook. However, that was not safe: a mutator
could add a step after that hook and it would fail to run as it could
try to connect to the node that was just decommissioned.

Since the full decommissioning was only happening after the upgrade to
the `current` version, it is sufficient (and simpler) to run that step
outside the mixedversion framework, after the `Run` function returns.

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/131507)
<!-- Reviewable:end -->

Commit:
roachtest: connect to storage cluter when running decommission commands

These commands only work on the storage cluster, so make sure to
specify the correct port in case the commands run in a multi-tenant
environment.

Fixes: #131550

Release note: None

renatolabs commented:
I disabled `separate-process` deployments in `decommissions/mixed-versions` because it is susceptible to the same issue we saw in other tests: at some point, the tenant becomes unresponsive, and a simple `SHOW CLUSTER SETTING version` times out after 30s. We need to investigate that first before re-enabling.

renatolabs commented:
[5 runs on this branch](https://teamcity.cockroachdb.com/viewLog.html?buildId=17100762&).

TFTRs!

bors r=srosenberg,herkolategan
